### PR TITLE
Fixes to gateway template environment variables

### LINKED ISSUE #16945: Expose LABELS environment variable on gateway
As a follow up to https://github.com/istio/istio/issues/16916, we can determine information from node metadata instead of pod if we do that.

I think we can do this with the downward api stuff, but didn't try -- we need it in json format, so may need changes to the injector.

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/16945

* Sidecar injector had typo
* Gateway was missing labels + cluster id

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 1]
```diff
@@ -1,18 +1,16 @@
 {{- range $key, $spec := .Values }}
 {{- if ne $key "enabled" }}
 {{- if $spec.enabled }}
+
+{{- $labels := dict "release" $.Release.Name "chart" (include "gateway.chart" $) "heritage" $.Release.Service }}
+{{- $labels = merge $labels $spec.labels }}
```
Comment (morvencao) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: There is an helm issue that complains the `=` here, which has been fixed from helm `v2.13.0`, see: https://github.com/helm/helm/issues/4987

Comment (howardjohn) [reply install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: Ahh thanks! thats why it worked locally then.

We advertise helm 2.10 support so probably need to find an even dirtier hack to accomplish this

**Comment by morvencao** [comment]
v

**Review by morvencao — APPROVED** [review]

**Pull request merged**